.github/copilot-instructions.md

```diff
@@ -0,0 +1,252 @@
+├── seismic/          # Seismic modeling examples
+├── cfd/              # Computational fluid dynamics
+├── misc/             # Miscellaneous examples
+└── userapi/          # User API demonstrations
+
+scripts/               # Utility scripts
+docker/               # Docker configurations  
+benchmarks/           # Performance benchmarks and ASV configuration
+binder/               # Jupyter Binder configuration
+```
+
+### Key Configuration Files
+- `pyproject.toml`: Build configuration, dependencies, flake8 settings (max-line-length=90)
+- `pytest.ini`: Test configuration and options
+- `requirements*.txt`: Dependency specifications for different environments (mpi, nvidia, testing, optional)
+- `environment-dev.yml`: Conda environment specification (recommended for development)
+- `.github/workflows/`: CI/CD pipelines (13 comprehensive workflows)
+- `MANIFEST.in`: Package distribution file inclusion/exclusion rules
+
+## Continuous Integration and Validation
+
+### GitHub Workflows (All Must Pass)
+1. **pytest-core-nompi.yml**: Core functionality tests across Python 3.10-3.13, multiple GCC versions
+2. **pytest-core-mpi.yml**: MPI parallel execution tests  
+3. **pytest-gpu.yml**: GPU computation tests (OpenACC/CUDA)
+4. **flake8.yml**: Code style validation (90-char line limit, specific ignore rules)
+5. **examples.yml**: Integration tests via example execution
+6. **examples-mpi.yml**: MPI-based example validation
+7. **tutorials.yml**: Jupyter notebook validation
+8. **asv.yml**: Performance benchmarking and regression detection
+9. **docker-bases.yml**: Docker base image builds for testing environments
+10. **docker-devito.yml**: Devito-specific Docker image builds
+11. **pythonpublish.yml**: Package publishing to PyPI
+12. **release-notes.yml**: Automated release documentation
+13. **triggers.yml**: Workflow orchestration and triggers
+
+### Pre-commit Validation Steps
+```bash
+# 1. Lint code
+flake8 --builtins=ArgumentError .
+
+# 2. Run core tests
+pytest -k "not adjoint" -m "not parallel" --maxfail=5 tests/
+
+# 3. Check configuration
+python3 -c "from devito import configuration; print(configuration)"
+
+# 4. Test example (optional but recommended)
+python examples/seismic/acoustic/acoustic_example.py
+```
+
+### Test Execution Times
+- Core tests: 10-15 minutes
+- Full test suite: 30-45 minutes
+- Individual test files: 1-5 minutes
+- Linting: <1 minute
+- Examples: 5-10 minutes each
+
+## Development Guidelines
+
+### Code Style (Enforced by CI)
+- **Line length**: 90 characters maximum (configured in pyproject.toml)
+- **Indentation**: 4 spaces (no tabs)
+- **Imports**: Grouped (stdlib, third-party, devito) and alphabetically sorted
+- **Docstrings**: NumPy-style documentation required for public APIs
+- **Naming**: CamelCase classes, snake_case methods/variables
+
+### Commit Messages and PR Titles
+Use descriptive prefixes to categorize changes (based on recent repository patterns):
```

Comment:
Add link to commit guidlines

https://github.com/devitocodes/devito/wiki/Tags-for-commit-messages-and-PR-titles

Reply:
Add "Capitalize title" e.g `misc: Patch` not `misc: patch`